lib/affiliate_credits.rb

```diff
@@ -33,19 +33,23 @@ def log_event(affiliate, user, credit, event)
   def check_affiliate
     return if cookies[:ref_id].blank? || @user.invalid?
 
-    if sender = Spree::User.find_by_ref_id(cookies[:ref_id])
-      affiliate = sender.affiliates.where(affiliate_email: @user.email).first
+    check_affiliate_internal(cookies[:ref_id], @user)
 
-      if affiliate.nil? && Spree::Affiliate.where(affiliate_email: @user.email).empty?
-        affiliate = sender.affiliates.create(affiliate_email: @user.email)
-      end
+    cookies[:ref_id] = nil
+  end
+
+  def check_affiliate_internal(ref_id, user)
+    if sender = Spree::User.find_by_ref_id(ref_id)
+      affiliate = sender.affiliates.where(affiliate_email: user.email).first
+
+      if affiliate.nil? && Spree::Affiliate.where(affiliate_email: user.email).empty?
+        affiliate = sender.affiliates.create(affiliate_email: user.email)
+      end 
```

Comment:
I think you can use `first_or_create` here instead

Reply:
This logic was already in place and likely dates back to 10 years ago, I wouldn't touch it with a 10ft pole if I can avoid it 😆